react-app/App.js

```diff
@@ -36,31 +37,55 @@ __DEV__ &&
   })
 
 export default function App() {
-  useFonts({
-    NotoSans: require('./assets/fonts/NotoSans-Regular.ttf'),
-    NotoSansBold: require('./assets/fonts/NotoSans-Bold.ttf'),
-    NotoSansExtraBold: require('./assets/fonts/NotoSans-ExtraBold.ttf'),
-  })
-
   useEffect(() => {
     db.transaction((tx) => {
       // create a table
       tx.executeSql(
         `create table if not exists Record (_id integer primary key AUTOINCREMENT, timestamp integer not null, exercise_name text not null);`
       )
     })
+    loadFontsAsync()
   }, [])
 
   const [reset, resetTrigger] = useState(false)
+  const [fontsLoaded, setFontsLoaded] = useState(false)
+
+  // Fonts setup
+  async function loadFontsAsync() {
+    await Font.loadAsync({
+      NotoSans: require('./assets/fonts/NotoSans-Regular.ttf'),
+      NotoSansBold: require('./assets/fonts/NotoSans-Bold.ttf'),
+      NotoSansExtraBold: require('./assets/fonts/NotoSans-ExtraBold.ttf'),
+    })
+    setFontsLoaded(true)
+  }
+
+  // Navigation setup
+  const Stack = createNativeStackNavigator()
+  const Tab = createBottomTabNavigator()
+
+  const HomeStack = () => {
+    return (
+      <Tab.Navigator
+        screenOptions={{
+          headerShown: false,
+        }}
+        tabBar={(props) => <TabBar {...props} />}
+      >
+        <Tab.Screen name='home' component={Home} />
+        <Tab.Screen name='mine' component={Setting} />
+      </Tab.Navigator>
+    )
+  }
 
-  return (
+  return fontsLoaded ? (
```

Comment:
In what situation will font failed to load? and how we should handle that by then?

Reply:
The page will not render if fonts have not loaded completely, the logic is far down to the line 105

Reply:
I see. then next question will be under what situation will the fonts fail to load?

Reply:
The reason I am asking is I am worrying when it failed to load the font (maybe because of the network condition), we won't be able to open the app for the users. Can we somehow fallback to standard fonts for that case?

Reply:
Yep I understand, it's better has an app usable with system fonts than couldn't open the app at all.